### PR TITLE
first

### PR DESCRIPTION
**Pull request by voloshynamaryna11**

### CODE REVIEW

File: src/main/java/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<CinemaHall> getAll() {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<CinemaHall> getAllMoviesQuery = session
```
Comment (Vosker) [inline src/main/java/cinema/dao/impl/CinemaHallDaoImpl.java]:
```suggestion
            Query<CinemaHall> getAllCinemaHalls = session
```

File: src/main/java/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> findAvailableSessionsQuery = session
+                    .createQuery("FROM MovieSession WHERE movie.id = :movieId "
+                            + "AND showTime > :start AND showTime < :end",
```
Comment (Vosker) [inline src/main/java/cinema/dao/impl/MovieSessionDaoImpl.java]:
```suggestion
                            + "AND showTime between :start AND :end",
```

**Review by Vosker — COMMENTED** [review]
ok just fix naming a bit

File: src/main/java/cinema/model/MovieSession.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import javax.persistence.ManyToOne;
+
+@Entity
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne(cascade = CascadeType.ALL)
+    private Movie movie;
+    @ManyToOne(cascade = CascadeType.ALL)
```
Comment (RomanDubovskyi) [inline src/main/java/cinema/model/MovieSession.java]: cascadeType all is a bad practice, do we really need this property, why ?

File: src/main/java/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movieSession);
+            transaction.commit();
+            return movieSession;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't insert movie session Entity", e);
```
Comment (RomanDubovskyi) [inline src/main/java/cinema/dao/impl/MovieSessionDaoImpl.java]: let's make message more informative and add some info about entity we were trying to save same with other methods, where applicable

File: src/main/java/cinema/model/MovieSession.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+
+@Entity
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne(cascade = CascadeType.ALL)
```
Comment (RomanDubovskyi) [inline src/main/java/cinema/model/MovieSession.java]: this is for your info, but keep in mind that toOne relations have fetchType eager by default, which is not always quite good

**Comment by RomanDubovskyi** [comment]
travis failed

**Comment by RomanDubovskyi** [comment]
let's remember in future about informative commit messages 
image

File: src/main/java/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+            Query<MovieSession> findAvailableSessionsQuery = session
+                    .createQuery("FROM MovieSession WHERE movie.id = :movieId "
+                            + "AND showTime BETWEEN :start AND :end",
+                            MovieSession.class);
+            findAvailableSessionsQuery.setParameter("movieId", movieId);
+            findAvailableSessionsQuery.setParameter("start", date.atTime(LocalTime.MIN));
+            findAvailableSessionsQuery.setParameter("end", date.atTime(LocalTime.MAX));
+            return findAvailableSessionsQuery.getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't find available sessions", e);
```
Comment (RomanDubovskyi) [inline src/main/java/cinema/dao/impl/MovieSessionDaoImpl.java:30]: it's better to add query info to message as well (date, movieId)

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**